src/addDefaultProps.tsx

```diff
@@ -0,0 +1,31 @@
+import isFunction from './utils/isFunction'
+import addProps from './addProps'
+import {ValueOrFunctionOfProps, CurriedPropsAdder} from './helperTypes'
+
+type AddDefaultPropsType = <
+  TProps extends {},
+  // TAdditionalProps extends Partial<TProps>
+  TAdditionalProps
+>(
+  createProps: ValueOrFunctionOfProps<TAdditionalProps, TProps>,
```

Comment:
The `ValueOrFunctionOrProps` change sounds good to me.

> I don't think conceptually it makes sense for it to accept a dependencies argument

Agreed, it seems weird. Like you say, it might make sense if the defaults are very expensive to calculate, but that seems like a pretty weird use case.